lib/compile.ml

```diff
@@ -22,7 +22,8 @@ type[@warning "-69" (* unused-record-field *)] function_sig =
 
 type predicate_sig =
   { pred_iargs : (Sym.t * BaseTypes.t) list;
-    pred_output : BaseTypes.t
+    pred_output : BaseTypes.t;
+    pred_loc : Locations.t
```

Comment:
Can you rename this to `pred_output_loc`? Or better yet, change the type of `pred_output` to a pair of `Locations.t * BaseTypes.t`?